File: client/node/signIn.js
const {SignInRequest} = require("./users_pb.js");

/**
 * @param {!proto.pb.Authentication.prototype} client The
 *     rpc client
 * @param body Payload with sign in credentials
 * @param {!string} body.email User email
 * @param {!string} body.password User password
 * @param callback
 *     call metadata
 *     callback The callback function(response)
 */
module.exports = function signIn(client, body, callback) {
    const {
        email,
        password
    } = body

    let request = new SignInRequest();
    request.setEmail(email);
    request.setPassword(password);

    client.signIn(request, {}, function(err, response) {
        if (err) {
            return callback({
                status: {
                    code: err.code,
                    details: err.message,
                    metadata: err.metadata
                },
                data: null
            });
        }
        let user = response.getUser();
        return callback({
            status: {
                code: 0,
                details: "Sign in successfully",
                metadata: response.metadata
            },
            data: {
                User: {
                    Id: user.getId(),
                    Name: user.getName(),
                    Email: user.getEmail(),
                    Created: user.getCreated(),
                    Updated: user.getUpdated()
                },
                Token: response.getToken()
            }
        });
    });
}

File: client/go/signIn.go
package client

import (
	"context"
	"github.com/tomashilaire/ct-users/client/go/pb"
	"log"
	"time"
)

// SignIn calls sign up RPC
func (grpcClient *GrpcClient) SignIn(email string, password string) (user *pb.User, token string, err error) {
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()
	req := new(pb.SignInRequest)
	req.Email = email
	req.Password = password
	resp, err := grpcClient.authService.SignIn(ctx, req)
	if err != nil {
		log.Println("cannot sign in, error: ", err)
		return user, "", err
	}
	return resp.User, resp.Token, nil
}
